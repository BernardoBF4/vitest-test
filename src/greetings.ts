export function greetings(name: string) {
  console.log(`Hello, ${name}`)
}
