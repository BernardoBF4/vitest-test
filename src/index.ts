export function deepMerge(a: Object | Array<any>, b: Object | Array<any>) {
  if (Array.isArray(a) && Array.isArray(b)) {
    return [...a, ...b]
  }

  return Object.assign(a, b)
}
