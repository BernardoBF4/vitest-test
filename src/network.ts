export async function getPost(id: number) {
  const data = await fetch('https://jsonplaceholder.typicode.com/posts/1').then((res) => res.json())

  return data.body
}
