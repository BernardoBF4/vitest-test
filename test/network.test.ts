import { expect, test, vi } from 'vitest'
import { getPost } from '../src/network'

vi.stubGlobal('fetch', async () => {
  return {
    json() {
      return { body: 'foo' }
    },
  }
})

test('should fetch', async () => {
  const result = await getPost(1)
  expect(result).toMatchInlineSnapshot('"foo"')
})
