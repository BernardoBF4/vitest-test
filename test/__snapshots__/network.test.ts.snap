// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should fetch 1`] = `
"quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto"
`;
